docs/architecture/blazor-for-web-forms-developers/architecture-comparison.md

```diff
@@ -8,6 +8,26 @@ ms.date: 09/11/2019
 
 # Architecture comparison of ASP.NET Web Forms and Blazor
 
+While ASP.NET Web Forms and Blazor have many similar concepts, they also have difference in how they work. Here we will take a look at the underlying workings and architecture of ASP.NET Web Forms and Blazor to see how they compare.
+
 ## ASP.NET Web Forms
 
-## Blazor
\ No newline at end of file
+TODO
+
+## Blazor
+
+Blazor is client-side web UI framework similar in nature to JavaScript frontend frameworks like Angular or React. The Blazor framework deals with handling of user interactions and rendering UI updates accordingly. Blazor is *not* based on a request-reply model. User interactions are handled as events that are not in the context of any particular HTTP request.
+
+Blazor apps consist of one or more root components that are rendered on an HTML page.
+
+![Blazor components in HTML](./media/architecture-comparison/blazor-components-in-html.png)
+
+How the user specifies where components should render and how the components are then wired up for user interactions is [hosting model](./hosting-models) specific.
+
+Blazor [components](./components) are .NET classes that represent a reusable piece of UI. Each component maintains its own state and specifies its own rendering logic, which can include rendering other components. Components specify event handlers for specific user interactions to update the component's state.
```

Comment:
I assume there's only C# support at this time. Should you say "C# classes" instead of ".NET classes"?

Reply:
We do generate C# classes from the .razor files as part of compiling them, but the end product is a compiled type in a .NET assembly, which is not language specific.